pyproject.toml

```
@@ -108,15 +99,47 @@ skips = [
 ignore-words-list = "elemt"
 skip = "./.git,./install,./build,./cmake-build*,./lib/cmake/sanitizers-cmake,./.tox,*/*_cache,*/.virtual_documents,*/.ipynb_checkpoints,*.pdf,*.svg"
 
-[tool.black]
-# Runs into a bug with this file. But since it is autogenerated, we don't want to reformat it anyway.
-force-exclude = "src/scipp/core/cpp_classes.pyi"
-skip-string-normalization = true
+[tool.ruff]
+line-length = 88
+extend-include = ["*.ipynb"]
+extend-exclude = [
+    ".*", "__pycache__", "cmake-build-*", "dist",
+    # Autogenerated, so no need to reformat:
+    "src/scipp/core/cpp_classes.pyi"]
+
+[tool.ruff.lint]
+select = ["B", "E", "F", "G", "I", "S", "T20", "PGH", "RUF"]
+ignore = [
+    "E111", "E114", "E117", "D206", "D300",  # conflict with ruff format
+]
+fixable = ["I001", "B010"]
+isort.known-first-party = ["scipp"]
+pydocstyle.convention = "numpy"
+
+[tool.ruff.lint.per-file-ignores]
+"src/scipp/__init__.py" = ["I"]
+"src/scipp/core/__init__.py" = ["I"]
```

```suggestion
"__init__.py" = ["I"]
```

should ignore in all `__init__.py` files.

---

Why? Most `__init__.py` files are simple and can be safely reordered.

---

I think it's just a good practice and we can generalize this upstream then, sorting `__init__.py` is usually a major source of random import errors.